### DISCUSSION TITLE
How we can persists schedules across restart of application?

### DISCUSSION THREAD

**Discussion by madhu72** [go-co-op/gocron#139] [Q&A]
There is always possible that, the application needs to restart, what is the possibility that keeps the schedules run across these restarted sessions?

**Comment by JohnRoesler** [comment]
Hello @madhu72, you would want to stop and serialize the scheduler on shutdown to some persistent storage and then de-serialize on start up of your application and start the scheduler again.

Take a look at 	https://stackoverflow.com/questions/28020070/golang-serialize-and-deserialize-back regarding serializing a go struct.

**Reply by madhu72** [reply]
Thank you very much for me to point to GOB @JohnRoesler

**Comment by madhu72** [comment]
@JohnRoesler aa the gocron scheduler does not containing exported fields it is not possible to serialize it. Please advice on this?

**Reply by JohnRoesler** [reply]
Ah, good point. We would need to add methods to the scheduler to expose a serialized scheduler and take in a serialized scheduler as the fields are not exported.
